tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -193,6 +196,24 @@ public IScope ActiveScope
         /// </summary>
         public string DefaultServiceName => TracerManager.DefaultServiceName;
 
+        /// <summary>
+        /// Gets the local root span of the current active trace
+        /// </summary>
+        internal ISpan LocalRootSpan
+        {
+            get
+            {
+                if (ScopeManager.Active == null || ScopeManager.Active.Span is not Span span)
```

Comment:
I'm not sure that using `ScopeManager.Active` is correct. It skips the call to `DistributedTracer.Instance.GetActiveScope()` in `Tracer.ActiveScope` used to resolve version mismatch issues.

https://github.com/DataDog/dd-trace-dotnet/blob/95f5448e178aa72e2377b5e5bf0d1636710846d1/tracer/src/Datadog.Trace/Tracer.cs#L168-L174

Unless I'm missing something, I think we can implement `LocalRootSpan` like this:
```csharp
get
{
    return (ActiveScope as Scope)?.Span.Context.TraceContext.RootSpan;
}
```